### PR TITLE
Luke/dpe 1550 common spread logic and tests

### PR DESCRIPTION
**Pull request by lukecaan**

### CODE REVIEW

File: common-ts/src/utils/math.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { BN, L2OrderBook, PERCENTAGE_PRECISION } from '@drift-labs/sdk';
+
+const calculatemarkPrice = (bestBidPrice: BN, bestAskPrice: BN) => {
```
Comment (lowkeynicc) [inline common-ts/src/utils/math.ts:3]: nitpick but this and the function below should be Mark to keep the camelcase consistent

**Review by lowkeynicc — APPROVED** [review]
just had a small nit about the camelcase, can't see any other issues

File: common-ts/package.json [hunk 2]
```diff
@@ -34,6 +34,8 @@
     "node": ">=14 <19"
   },
   "devDependencies": {
+	"@types/mocha": "^10.0.4",
```
Comment (crispheaney) [inline common-ts/package.json:37]: nit: indenting seems off

File: common-ts/src/utils/math.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	return bestBidPrice.add(bestAskPrice).divn(2);
```
Comment (crispheaney) [inline common-ts/src/utils/math.ts:4]: do you know if this is what we were using before? @0xbigz you should review this, not sure what you had when it was crossing before

File: common-ts/src/utils/math.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+	return {
+		bestBidPrice,
+		bestAskPrice,
+		markPrice,
+	};
+};
+
+const calculateSpreadQuote = (bestBidPrice: BN, bestAskPrice: BN) => {
+	return bestBidPrice.sub(bestAskPrice).abs();
```
Comment (crispheaney) [inline common-ts/src/utils/math.ts:28]: should have @0xbigz look at this, if they're crossing we might want it to show 0

Comment (lowkeynicc) [reply common-ts/src/utils/math.ts:28]: actually yea i think we showed negative before? confirm with z

Comment (0xbigz) [reply common-ts/src/utils/math.ts:28]: is showing negative bad? assuming now dlob construction/render issues, if theres actually that level of spread its represents makers subsidizing their maker rebate. also be nice to know if roundtrip exceeds taker fee in arbitrage cases

**Pull request merged**